src/main/scala/calc/Compiler.scala

```diff
@@ -74,6 +88,71 @@ object Compiler {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
+      case BinaryOp(op, rhs, lhs) =>
+        val right = compileExpr(rhs, listId)
+        val left  = compileExpr(lhs, listId)
+        irt.BinaryOp(binaryOpMap(op), right, left)
+
+      case Let(name, value, body) =>
+        val irtId = irt.Ident(name.name)
+        val letVal = compileExpr(value, listId)
+        irt.Block(List(
+          irt.VarDef(irtId, letVal.tpe, false, letVal),
+          compileExpr(body, listId + (name.name -> letVal.tpe)))
+        )
+
+      case Ident(name) =>
+        listId.contains(name) match {
+          case true  => irt.VarRef(irt.Ident(name))(listId(name))
+          case false => throw new Exception(s"Identifier ${name} not defined")
+        }
+
+      case If(cond, thenp, elsep) =>
+        val ifVal = compileExpr(cond, listId)
+        irt.If(
+          irt.BinaryOp(Num_!=, ifVal, irt.DoubleLiteral(0)),
+          compileExpr(thenp, listId),
+          compileExpr(elsep, listId)
+        )(irtpe.DoubleType)
+
+      case Closure(params, body) =>
+        //Argument unboxing at the start of the function
+        val listPar = params map { x =>
+                        irt.ParamDef(irt.Ident("_" + x.name), irtpe.AnyType,
+                          false, false)
+                      }
+        val stVar = params map { x =>
+                      irt.VarDef(irt.Ident(x.name), irtpe.DoubleType, false,
+                        irt.Unbox(irt.VarRef(irt.Ident("_" + x.name))
+                          (irtpe.AnyType), 'D')
+                      )
+                    }
+        //For the capture list
+        val capList = listId.toList
+        val capDef = capList.map { case (x,y) => irt.ParamDef(irt.Ident(x), y,
+          false, false)}
+        val capVal = capList.map { case (x,y) => irt.VarRef(irt.Ident(x))(y)}
+        //The body of the closure
+        val typePar = params map ( x => (x.name, irtpe.DoubleType))
+        val bl = irt.Block(stVar ++ List(compileExpr(body, listId ++ typePar)))
+
+        irt.Closure(capDef, listPar, bl, capVal)
+
+      case Call(Ident(name) , args) if !(listId.contains(name)) =>
+        val argList = args.map(x => compileExpr(x,listId))
+        val mathModule = irt.LoadModule(mathClass)
+        val fun = irt.Ident(nativeFun(name))
+
+        nativeFun.contains(name) match {
+          case true  => irt.Apply(mathModule, fun, argList)(irtpe.DoubleType)
+          case false => throw new Exception(s"Function ${name} not found")
```

Comment:
Speaking of `case true`, that's bad style. Use an `if (nativeFun.contains(name))` instead.